### PR TITLE
Updates styling on challenge card

### PR DESCRIPTION
**Pull request by mk4111**
See #668 

Styling on challenge card for challenges list view, updated.
As the .card and .card--challenge classes are shared in other places, have manually gone through the screens to check for unwanted differences. Don't think I came across any, but if you do, please let me know! Thanks!

### CODE REVIEW

File: templates/views/organisations/details.html [hunk 1]
```diff
@@ -83,7 +83,6 @@ <h2 class="desktop card card__header card--desktop-header-bottom-border">Our Cha
 
         {{#if permissions.primary}}
           {{#if permissions.editable}}
-            <a href="/challenges/add" class="card__copy card__additional float-left card__data--link">Add challenge</a>
```
Comment (mk4111) [inline templates/views/organisations/details.html]: This deleted line isn't relevant to #685 but I don't think it should be here at all, so have removed

File: public/css/main.css [hunk 2]
```diff
@@ -634,10 +639,12 @@ Light-teal:     #66d9e0;
     .card__additional {
       color: #89a0ae;
       transform: translate(0%,100%);
-      text-align: right;
       -webkit-transform: translate(30deg); /* Ch <36, Saf 5.1+, iOS < 9.2, An =<4.4.4 */
       -ms-transform: translate(30deg); /* IE 9 */
     }
+        .card__additional--aqua {
```
Comment (JMurphyWeb) [inline public/css/main.css]: Not sure if you have seen the class .anchor--teal? it does exactly this. if it doesn't work, do you think we could stick to the '.card__additional--teal' for consistency?

File: templates/partials/challenge.html [hunk 3]
```diff
@@ -12,8 +12,8 @@
   </a>
 
   {{#if shared_by}}
-  <p class="card__copy card__additional">Shared by
-    <a href="/orgs/{{org_id}}" class="card__data--link">
+  <p class="card__copy card__additional u-teal">Shared by
+    <a href="/orgs/{{org_id}}" class="card__data--underline">
```
Comment (harrygfox) [inline templates/partials/challenge.html]: I'd be a little wary of introducing to many names that reference the specific styles the enact e.g. colours, font decoration etc. Seems okay _**for now**_ but just be aware of that

File: templates/views/insights/add.html [hunk 4]
```diff
@@ -15,7 +15,7 @@ <h1 class="snackbar-content snackbar-content--error">{{error.message}}</h1>
     </div>
     {{else}}
     <div class="snackbar snackbar--success">
-      <h1 class="snackbar-content snackbar-content--success">have you uploaded your asset? If not do it <a class="anchor anchor--teal" href="#">here</a></h1>
+      <h1 class="snackbar-content snackbar-content--success">have you uploaded your asset? If not do it <a class="anchor u-teal" href="#">here</a></h1>
```
Comment (harrygfox) [inline templates/views/insights/add.html]: should this not be 
`snackbar__content` ?
or is snackbar-content a separate block?

File: templates/partials/challenge.html (hunk 3 above)
Comment (mk4111) [inline templates/partials/challenge.html]: Yeah, that's a good point - I'll be sure to be careful and use it sparingly!! May need to change the classname later on, but will keep that in mind!

File: templates/views/insights/add.html (hunk 4 above)
Comment (mk4111) [inline templates/views/insights/add.html]: yup snackbar-content is a block in itself

File: templates/partials/challenge.html (hunk 3 above)
Comment (mk4111) [inline templates/partials/challenge.html]: Although that said, we do now have the `.u-teal` class :thinking: :grinning:

**Comment by mk4111** [comment]
Sorry for the extra hassle @JMurphyWeb and @harrygfox !!
I made a wee update on the way we we were displaying items in the `explore` view because there was some discrepancy between orgs + people lists vs. challenges and insights/resources list (please see #681 for a better description!) If you could let me know what you think that'd be grand! Thanks gaiiizzz <3

File: templates/views/insights/add.html (hunk 4 above)
Comment (JMurphyWeb) [inline templates/views/insights/add.html]: @mk4111 I think snackbar__content makes sense. To me, the block is the snackbar, elements inside that block are the content and maybe an icon if we decide to add one later.

**Comment by JMurphyWeb** [comment]
@mk4111 https://github.com/emfoundation/ce100-app/pull/685#discussion_r101745278

File: templates/views/insights/add.html (hunk 4 above)
Comment (mk4111) [inline templates/views/insights/add.html]: That makes sense - let's change it to that! Shall we make that a separate PR because it makes changes to a bunch of other forms?

Comment (mk4111) [inline templates/views/insights/add.html]: Gonna update in this PR!

**Comment by harrygfox** [comment]
♻️ back to you @mk4111 ♻️ 
for tests and conflicts

**Comment by mk4111** [comment]
@harrygfox , back to youuu pummmpkin :musical_note:

File: templates/partials/challenge.html [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
+    <a href="/challenges/{{id}}" class="card__header--challenge card__data--link">
+      {{title}}
+      <i class="navbar__icon icon icon--open-link"></i>
     </a>
-  </p>
-  {{/if}}
-</div>
+
+    {{#if shared_by}}
+    <p class="card__copy card__additional u-teal">Shared by
```
Comment (harrygfox) [inline templates/partials/challenge.html]: really not sure about `card__additional`: in this use case (with the `u-teal`), it does nothing but transform itself down by half - which to my mind is something a _utility class would do_. hmm

**Firstly** I'd advise you make card__additional into `card__copy--additional` and make it _just a colour declaration_ (i see from the other use cases in the app that the teal colour is a one-of-a-kind). 

**Secondly**: the need to move this element down by half it's own height. 
- If this is something you'll need to do in other places, create a utility class `u-transformY50` or something. 
- If this is specific to these additional bits of copy in the cards, then really you should just use margin on the top **and** bottom. so `.card__copy` would become `{ margin: 1rem auto; }`. 
    - **remember** if you do _this latter option_ of adding margin to `card__copy` be mindful of situations in which you have card__copy nested within card__copy - the margin will double which is not the desired effect (e.g. in `views/challenges/details.html` lines 28+29)

File: templates/views/organisations/details.html [hunk 6]
```diff
@@ -83,8 +83,8 @@ <h2 class="desktop card card__header card--desktop-header-bottom-border">Our Cha
...
-            <a href="/orgs/{{org.id}}/archived-challenges" class="card__copy card__additional float-right card__data--link">Go to archived challenges</a>
+             <a href="/challenges/add" class="card__copy card__additional float-left card__data--link">Add challenge</a>
```
Comment (harrygfox) [inline templates/views/organisations/details.html]: wo there `float-left` 👀  shouldn't you have some namespace in front of your sweet `f`. @JMurphyWeb wasssuuuup. you know what i'm sayin

File: templates/partials/challenge.html (hunk 5 above)
Comment (mk4111) [inline templates/partials/challenge.html]: @harrygfox - I understand what you mean. Having made the relevant changes in [this PR](https://github.com/emfoundation/ce100-app/pull/679#discussion_r102058946), I realised the `anchor__shared-by` could and probably should be applied here too for the sake of consistency, so that's what I've done. Hope this is ok!

File: templates/views/organisations/details.html (hunk 6 above)
Comment (mk4111) [inline templates/views/organisations/details.html]: Haaai

File: templates/partials/insights.html [hunk 7]
```diff
@@ -5,6 +5,6 @@
     </a>
   {{/if}}
   <a href={{url}} target="_blank" class="card__header card__header--challenge card__data--link">{{title}}</a>
-  <p class="card__copy card__additional">By {{author}}</p>
-  <p class="card__copy card__additional card__additional--small">{{type}}</p>
+  <p class="anchor anchor__shared-by--left">By {{author}}</p>
+  <p class="anchor anchor__shared-by--left anchor__shared-by--small">{{type}}</p>
```
Comment (JMurphyWeb) [inline templates/partials/insights.html]: Think these need the element `anchor__shared-by` before the modifier?

File: templates/views/organisations/details.html [hunk 8]
```diff
@@ -83,8 +83,8 @@ <h2 class="desktop card card__header card--desktop-header-bottom-border">Our Cha
 
         {{#if permissions.primary}}
           {{#if permissions.editable}}
-             <a href="/challenges/add" class="card__copy card__additional float-left card__data--link">Add challenge</a>
-             <a href="/orgs/{{org.id}}/archived-challenges" class="card__copy card__additional float-right card__data--link">Go to archived challenges</a>
+             <a href="/challenges/add" class="anchor anchor__shared-by--left u-float-left">Add challenge</a>
+             <a href="/orgs/{{org.id}}/archived-challenges" class="anchor anchor__shared-by--left u-float-right">Go to archived challenges</a>
```
Comment (JMurphyWeb) [inline templates/views/organisations/details.html]: Again, not sure if this is correct, but my understanding was we need an element in order to use a modifier of that element? Worth checking with @harrygfox first.

File: templates/partials/insights.html (hunk 7 above)
Comment (mk4111) [inline templates/partials/insights.html]: Yeah maybe! I thought modifiers could go on the block or the element, but I might be wrong!

Comment (harrygfox) [inline templates/partials/insights.html]: The issue here is that you have a _block_ and an _element_ sharing the same DOM node. It looks to me like anchor__shared-by is a _**kind**_ of anchor rather than a dependent sub-component of anchor.
perhaps you should separate out modifiers:
`anchor` `anchor--shared-by` `anchor--left` `anchor--small`

**Comment by harrygfox** [comment]
@mk4111 The above comment on this PR is the only qualm I have with it. Then it's good to GO!

File: templates/partials/insights.html (hunk 7 above)
Comment (harrygfox) [inline templates/partials/insights.html]: @mk4111 you're both right. the problem is that this node is professing to be both an `anchor` plain and simply **and** a sub-set of `anchor`, the "anchor shared-by".... which isn't a noun, so semantically it doesn't work so well

**Comment by harrygfox** [comment]
@mk4111 ready to be merged once conflicts are resolved

**Comment by harrygfox** [comment]
@mk4111 moving target or what ey 🎯
